keyboards/minidox/keymaps/dustypomerleau/keymap.c

```diff
@@ -0,0 +1,490 @@
+// Minidox keymap by dustypomerleau
+// Thanks for checking out my keymap. The rationale behind the layout is described in the README.
+
+#include QMK_KEYBOARD_H
+
+extern keymap_config_t keymap_config;
+
+enum custom_layers {
+  _CMK_DHM,
+  _QWERTY,
+  _SYS,
+  _NAV,
+  _NUM_E,
+  _NUM_N,
+  _SYM
+};
+
+enum custom_keycodes {
+  CMK_DHM = SAFE_RANGE,
+  QWERTY,
+  SYS,
+  NAV,
+  NUM_E,
+  NUM_N,
+  SYM
+};
+
+typedef struct {
+  bool is_press_action;
+  int state;
+} tap;
+
+enum {
+  SINGLE_TAP = 1,
+  SINGLE_HOLD = 2,
+  DOUBLE_SINGLE_TAP = 3
+};
+
+enum {
+  ALT_OP = 0,
+  CTL_CCB,
+  GUI_CP,
+  SFT_OCB,
+  SFT_PLS
+};
+
+int cur_dance (qk_tap_dance_state_t *state);
+void altop_finished (qk_tap_dance_state_t *state, void *user_data);
+void altop_reset (qk_tap_dance_state_t *state, void *user_data);
+void ctlccb_finished (qk_tap_dance_state_t *state, void *user_data);
+void ctlccb_reset (qk_tap_dance_state_t *state, void *user_data);
+void guicp_finished (qk_tap_dance_state_t *state, void *user_data);
+void guicp_reset (qk_tap_dance_state_t *state, void *user_data);
+void sftocb_finished (qk_tap_dance_state_t *state, void *user_data);
+void sftocb_reset (qk_tap_dance_state_t *state, void *user_data);
+void sftpls_finished (qk_tap_dance_state_t *state, void *user_data);
+void sftpls_reset (qk_tap_dance_state_t *state, void *user_data);
+
+#define _______ KC_TRNS
+#define ALT_2 LALT_T(KC_2)
+#define ALT_3 LALT_T(KC_3)
+#define ALT_8 LALT_T(KC_8)
+#define ALT_D LALT_T(KC_D)
+#define ALT_E LALT_T(KC_E)
+#define ALT_K LALT_T(KC_K)
+#define ALT_OB LALT_T(KC_LBRC)
+#define ALT_S LALT_T(KC_S)
+#define CTRL_2 LCTL_T(KC_2)
+#define CTRL_4 LCTL_T(KC_4)
+#define CTRL_5 LCTL_T(KC_5)
+#define CTRL_9 LCTL_T(KC_9)
+#define CTRL_EQ LCTL_T(KC_EQL)
+#define CTRL_I LCTL_T(KC_I)
+#define CTRL_L LCTL_T(KC_L)
+#define CTRL_R LCTL_T(KC_R)
+#define CTRL_S LCTL_T(KC_S)
+#define GUI_0 LGUI_T(KC_0)
+#define GUI_1 LGUI_T(KC_1)
+#define GUI_4 LGUI_T(KC_4)
+#define GUI_7 LGUI_T(KC_7)
+#define GUI_CB LGUI_T(KC_RBRC)
+#define GUI_F LGUI_T(KC_F)
+#define GUI_J LGUI_T(KC_J)
+#define GUI_N LGUI_T(KC_N)
+#define GUI_TEA LGUI_T(KC_T)
+#define MAC_EM S(LALT(KC_MINS))
+#define MAC_EN LALT(KC_MINS)
+#define NAV_BK LT(_NAV, KC_BSPC)
+#define NAV_LK TG(_NAV)
+#define NUME_SPC LT(_NUM_E, KC_SPC)
+#define NUMLK_E TG(_NUM_E)
+#define NUMLK_N TG(_NUM_N)
+#define NUMN_SPC LT(_NUM_N, KC_SPC)
+#define SFT_0 LSFT_T(KC_0)
+#define SFT_1 LSFT_T(KC_1)
+#define SFT_6 LSFT_T(KC_6)
+#define SFT_7 LSFT_T(KC_7)
+#define SFT_A LSFT_T(KC_A)
+#define SFT_O LSFT_T(KC_O)
+#define SFT_OS OSM(MOD_LSFT)
+#define SFT_QOT LSFT_T(KC_QUOT)
+#define SYM_OS OSL(_SYM)
+#define SYS_Z LT(_SYS, KC_Z)
+#define VOL_DN S(LALT(KC__VOLDOWN))
+#define VOL_UP S(LALT(KC__VOLUP))
+#define XXXXXXX KC_NO
```

Comment:
Line 106 Not needed if you use QMK_KEYBOARD_H

Reply:
These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```